docs/docs/reference/contextual/derivation.md

```diff
@@ -364,6 +379,35 @@ Assuming the `Ordering.derived` method has a given parameter of type `Mirror[T]`
 compiler generated `Mirror` instance for `Option` and the derivation of the instance will be expanded on the right
 hand side of this definition in the same way as an instance defined in ADT companion objects.
 
+### Automatic Derivation
+
+Automatic derivation of Typeclass instances is less efficient than explicit derivation since it requires a derivation
+to be performed at every site it is invoked. For many use-cases however, this is sufficient. The `derived` appraoch
+above allows for the following implementation.
+
+```scala
+object AutomaticEq {
+  inline implicit def autoEq[T]: Eq[T] = Eq.derived
+  inline def [T](a: T) === (b: T) = autoEq[T].eqv(a, b)
+}
+```
+It is then possible to use the `Eq` Typeclass without explicit specification of `Eq` givens or `derives` clauses.
+Using our simple ADT defined above:
+```scala
+enum Opt[+T] { // derives Eq - automatically inferred
```

Comment:
I would move the comment to the call sites

Reply:
Sure